crates/red_knot_python_semantic/resources/mdtest/assignment/annotations.md

```diff
@@ -23,12 +23,11 @@ x: int
 x = "foo"  # error: [invalid-assignment] "Object of type `Literal["foo"]` is not assignable to `int`"
 ```
 
-## PEP-604 annotations not yet supported
+## PEP-604 annotations are supported
 
 ```py
-def f() -> str | None:
+def f() -> str | int | None:
     return None
```

Comment:
Could you also add a test that uses a union with duplicate elements, e.g. `str | int | str | None`? We should infer the type as `str | int | None` (we should deduplicate the elements)